src/components/Pressable/GenericPressable/GenericPressable.js

```diff
@@ -0,0 +1,149 @@
+        return styles.cursorDisabled;
+    }
+    if (isText) {
+        return styles.cursorText;
+    }
+    return styles.cursorPointer;
+};
+
+const GenericPressable = forwardRef((props, ref) => {
+    // eslint-disable-next-line react/destructuring-assignment
+    const {
+        children,
+        onPress,
+        onLongPress,
+        onKeyPress,
+        disabled,
+        style,
+        accessibilityHint,
+        shouldUseHapticsOnLongPress,
+        shouldUseHapticsOnPress,
+        nextFocusRef,
+        keyboardShortcut,
+        shouldUseAutoHitSlop,
+        enableInScreenReaderStates,
+        ...rest
+    } = props;
+
+    const isScreenReaderActive = Accessibility.useScreenReaderStatus();
+    const [hitslop, onLayout] = Accessibility.useAutoHitSlop();
+
+    const isDisabled = useMemo(() => {
+        let shouldBeDisabledByScreenReader = false;
+        switch (enableInScreenReaderStates) {
+            case 'all':
+                shouldBeDisabledByScreenReader = false;
+                break;
+            case 'disabled':
+                shouldBeDisabledByScreenReader = !isScreenReaderActive;
+                break;
+            case 'active':
+                shouldBeDisabledByScreenReader = isScreenReaderActive;
+                break;
+            default:
+                break;
+        }
+        return props.disabled || shouldBeDisabledByScreenReader;
+    }, [isScreenReaderActive, enableInScreenReaderStates, props.disabled]);
+
+    const onLongPressHandler = useCallback(() => {
+        if (!onLongPress) {
+            return;
+        }
+        if (shouldUseHapticsOnLongPress) {
+            HapticFeedback.longPress();
+        }
+        onLongPress();
+        Accessibility.moveAccessibilityFocus(nextFocusRef);
+    }, [shouldUseHapticsOnLongPress, onLongPress, nextFocusRef]);
+
+    const onPressHandler = useCallback(() => {
+        if (shouldUseHapticsOnPress) {
+            HapticFeedback.press();
+        }
+        onPress();
+        Accessibility.moveAccessibilityFocus(nextFocusRef);
+    }, [shouldUseHapticsOnPress, onPress, nextFocusRef]);
+
+    const onKeyPressHandler = useCallback((event) => {
+        if (event.key !== 'Enter') {
+            return;
+        }
+        onPressHandler();
+    }, [onPressHandler]);
+
+    useEffect(() => {
+        if (!keyboardShortcut) {
+            return;
+        }
+        const {shortcutKey, descriptionKey, modifiers} = keyboardShortcut;
+        const unsubscribe = KeyboardShortcut.subscribe(shortcutKey, onPressHandler, descriptionKey, modifiers, true, false, 0, false);
+        return () => {
+            if (!unsubscribe) {
+                return;
+            }
+            unsubscribe();
+        };
+    }, [keyboardShortcut, onPressHandler]);
+
+    return (
+        <Pressable
+            hitSlop={shouldUseAutoHitSlop && hitslop}
+            onLayout={onLayout}
+            ref={ref}
+            focusable
+            accessible
```

Comment:
This prop [isn't supported on web](https://necolas.github.io/react-native-web/docs/accessibility/#keyboard-focus):

> The tabIndex prop determines whether a component is user-focusable and appears in the keyboard tab flow. This prop should be used instead of the accessible (or focusable prop) found in React Native for Android/iOS, which is not implemented by React Native for Web/Windows/macOS.

So I don't think we should use it in the base implementation

Reply:
I didn't want to scatter the accessibility props across the platform-specific files. I was thinking that `accessible` `focusable` even as they are not supported on web, they won't be breaking anything. Same with `tabIndex`, passing it on native build shouldn't introduce problems.

Reply:
Ok, I guess it works both ways so NAB